django_enum_choices/fields.py

```diff
@@ -36,27 +38,29 @@ def __init__(self, enum_class: Enum, **kwargs):
             EnumValueMaxLengthValidator(kwargs['max_length'])
         )
 
-    def _pack_choices(self) -> Tuple[Tuple[Union[int, str]]]:
-        return [
-            (str(choice.value), str(choice.value))
-            for choice in self.enum_class
-        ]
+    def _get_choice_builder(self):
+        choice_builder = getattr(self.enum_class, 'choice_builder', value_value)
 
-    def build_choices(self):
-        get_readable_value = getattr(self.enum_class, 'get_readable_value', None)
+        if not callable(choice_builder):
+            raise EnumChoiceFieldException(
+                _('`{}.choice_builder` must be a callable.'.format(
+                    self.enum_class.__name__
+                ))
+            )
 
-        if callable(get_readable_value):
-            # Used if different readable values are wanted from the
-            # ones stored in the enum class
-            packed_choices = [(
-                str(choice.value), self.enum_class.get_readable_value(choice)
-            ) for choice in self.enum_class]
+        return as_choice_builder(choice_builder)
 
-            return packed_choices
+    def _value_from_built_choice(self, built_choice):
+        if isinstance(built_choice, tuple):
+            return built_choice[0]
 
-        packed_choices = self._pack_choices()
+        return built_choice
 
-        return packed_choices
+    def build_choices(self) -> Tuple[Tuple[str]]:
```

Comment:
Lets actually add validation that all elements are being returned as strings :+1: